### PR TITLE
Update to use a venv during software install

### PR DESCRIPTION
**Pull request by klaasnicolaas**
PR to update the documentation for using a virtual environment, while installing the RotorHazard software. This adjustment is due to the [PEP 668](https://peps.python.org/pep-0668/) message that you receive when installing the packages.

One thing still needs to be tested (open call to everyone):
- Using the LED Strip with a pin other than GPIO10 (SPI method), where you have to run the server as root user.

## Breaking changes:
- From RH4 onwards the recommended way is to create a venv, so you must create it as in the installation instructions.
- Users must adjust the path in their rotorhazard Linux service

### CODE REVIEW

**Comment by HazardCreative** [comment]
Closes #792

File: doc/Software Setup.md [hunk 1]
```diff
@@ -107,23 +107,28 @@ sudo update-alternatives --install /usr/bin/python python /usr/bin/python3 2
...
 ```
 
-Enter the commands below to install RotorHazard server dependencies (be patient, this may take a few minutes):
-```
+The next step is to create a virtual environment, to install the RotorHazard server dependencies (be patient, this may take a few minutes):
+
+```bash
 cd ~/RotorHazard/src/server
-sudo pip install -r requirements.txt
+python3 -m venv venv
```
Comment (ethomas997) [inline doc/Software Setup.md:129]: We're using "python" (not "python3") everywhere else in the doc; can this be "python -m venv venv"?  Will this work okay under Python 3.7?

Comment (klaasnicolaas) [reply doc/Software Setup.md:129]: Yes of course, I thought rotorhazard still uses python3 everywhere. But `python -m venv venv` would also be my preference 😉

**Comment by SpencerGraffunder** [comment]
This line could be updated

```
Note: if username was changed during Pi OS setup, be sure to change pi in User and WorkingDirectory to match your username.
```

to 

```
Note: if username was changed during Pi OS setup, be sure to change pi in User, WorkingDirectory, and ExecStart to match your username.
```

**Comment by SpencerGraffunder** [comment]
Does anything need to be done to re-setup the venv after an update as described in the docs? If so, the 'updating' section needs updating.

**Comment by klaasnicolaas** [comment]
You're right, the documentation for updating RH also needs to be updated. Because we are replacing the current folder with a new version, you also have to create a completely new venv 🤔 (keep the old one and just update it would be nicer)

**Comment by HazardCreative** [comment]
What's the solution? Adjust the Update section of the notes on future releases and require reinstall of dependencies each time? Or is it better to install the venv outside the RH folder?

**Comment by klaasnicolaas** [comment]
Copying the venv does indeed seem like a good solution, but are we going to use `~` everywhere or not? Because of the `cd ~` it would not really be necessary to apply that further in the commands.

```bash
cd ~
wget https://codeload.github.com/RotorHazard/RotorHazard/zip/1.2.3 -O temp.zip
unzip temp.zip
mv RotorHazard RotorHazard.old
mv RotorHazard-1.2.3 RotorHazard
rm temp.zip
cp RotorHazard.old/src/server/{config.json,database.db} RotorHazard/src/server/
cp RotorHazard.old/src/server/venv RotorHazard/src/server/
```

**Pull request merged**